org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/telemetry/DocumentTelemetryInfo.java

```diff
@@ -90,18 +87,17 @@
 			}
 
 			if (resolvers.size() == 1) {
-				props.put(DOC_PROP_RESOLVER, resolvers.iterator().next());
+				cache.put(String.join(".", DOC_PROP_RESOLVER, resolvers.iterator().next()));
 			} else {
-				props.put(DOC_PROP_RESOLVER, resolvers);
+				cache.put(String.join(".", DOC_PROP_RESOLVER, resolvers.toString()));
 			}
 			if (!identifiers.isEmpty()) {
 				if (identifiers.size() == 1) {
-					props.put(DOC_PROP_IDENTIFIER, identifiers.iterator().next());
+					cache.put(String.join(".", DOC_PROP_IDENTIFIER, identifiers.iterator().next()));
 				} else {
-					props.put(DOC_PROP_IDENTIFIER, identifiers);
+					cache.put(String.join(".", DOC_PROP_IDENTIFIER, identifiers.toString()));
 				}
 			}
 		}
```

Comment:
I think we can simplify this quite a bit at the cost of some data, that is arguably not as important. Instead of storing multiple identifiers, or resolvers, as an array to signifiy that one file had multiple of these, we can just aggregate them individually. This makes it easier to count files with more than one identifier/resolver, but harder to identify what are the popular combinations.

Reply:
Done.